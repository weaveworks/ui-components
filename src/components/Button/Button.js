import React from 'react';
import PropTypes from 'prop-types';

/**
 * A button that will run a callback on click
 * ```javascript
 *  <Button onClick={alert} text="Submit" />
 * ```
 */
import classnames from 'classnames';

class Button extends React.Component {
  constructor(props, context) {
    super(props, context);

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(ev) {
    if (this.props.onClick) {
      this.props.onClick(ev, this.props.text);
    }
  }

  render() {
    const { children, className, text, disabled, style, primary } = this.props;
    const cl = classnames('weave-button', { primary });
    return (
      <button
        style={style}
        disabled={disabled}
        onClick={this.handleClick}
        className={className || cl}
      >
        {children || text}
      </button>
    );
  }
}

Button.propTypes = {
  /**
 * Callback that will be run when the button is clicked.
 */
  onClick: PropTypes.func,
  /**
 * Text that will be used as the button label.
 * If this props is provided, it will be passed back to the `onClick` handler
 */
  text: PropTypes.string,
  /**
 * Disable the button.
 */
  disabled: PropTypes.bool,
  /**
   * Render the button in $turquoise (useful for CTAs)
   */
  primary: PropTypes.bool
};

Button.defaultProps = {
  text: 'Submit'
};

export default Button;
