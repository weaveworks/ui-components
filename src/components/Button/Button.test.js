import React from 'react';
import { shallow } from 'enzyme';
import expect, { createSpy } from 'expect';

import Button from './Button';

describe('<Button />', () => {
  let button, spy;

  beforeEach(() => {
    spy = createSpy();
    button = shallow(
      <Button onClick={spy} />
    );
  });
  it('runs a callback', () => {
    button.simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});
