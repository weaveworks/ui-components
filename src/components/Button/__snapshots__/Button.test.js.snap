// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> accepts a style prop 1`] = `
.c0 {
  height: 36px;
  min-width: 90px;
  padding: 0 12px;
  border: 0;
  outline: none;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  background: #fff;
  color: #000;
  font-size: 0.875em;
  text-transform: uppercase;
  cursor: pointer;
}

.c0:hover {
  -webkit-transition: color .3s ease;
  transition: color .3s ease;
  background: #fafafc;
  color: #3c3c5b;
}

<button
  className="c0"
  disabled={undefined}
  onClick={[Function]}
  style={
    Object {
      "color": "blue",
    }
  }
  type={undefined}
>
  Submit
</button>
`;

exports[`<Button /> snapshots renders danger 1`] = `
.c0 {
  height: 36px;
  min-width: 90px;
  padding: 0 12px;
  border: 0;
  outline: none;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  background: #ff0000;
  color: #fff;
  font-size: 0.875em;
  text-transform: uppercase;
  cursor: pointer;
}

.c0:hover {
  -webkit-transition: color .3s ease;
  transition: color .3s ease;
  background: #b30000;
  color: #fff;
}

<button
  className="c0"
  disabled={undefined}
  onClick={[Function]}
  style={undefined}
  type={undefined}
>
  Danger
</button>
`;

exports[`<Button /> snapshots renders default 1`] = `
.c0 {
  height: 36px;
  min-width: 90px;
  padding: 0 12px;
  border: 0;
  outline: none;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  background: #fff;
  color: #000;
  font-size: 0.875em;
  text-transform: uppercase;
  cursor: pointer;
}

.c0:hover {
  -webkit-transition: color .3s ease;
  transition: color .3s ease;
  background: #fafafc;
  color: #3c3c5b;
}

<button
  className="c0"
  disabled={undefined}
  onClick={[Function]}
  style={undefined}
  type={undefined}
>
  Submit
</button>
`;

exports[`<Button /> snapshots renders disabled 1`] = `
.c0 {
  height: 36px;
  min-width: 90px;
  padding: 0 12px;
  border: 0;
  outline: none;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  background: #fafafc;
  color: #aaaaaa;
  font-size: 0.875em;
  text-transform: uppercase;
  cursor: not-allowed;
}

.c0:hover {
  -webkit-transition: color .3s ease;
  transition: color .3s ease;
  background: #fafafc;
  color: #aaaaaa;
}

<button
  className="c0"
  disabled={true}
  onClick={[Function]}
  style={undefined}
  type={undefined}
>
  Disabled
</button>
`;

exports[`<Button /> snapshots renders primary 1`] = `
.c0 {
  height: 36px;
  min-width: 90px;
  padding: 0 12px;
  border: 0;
  outline: none;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  background: #00bcd4;
  color: #fff;
  font-size: 0.875em;
  text-transform: uppercase;
  cursor: pointer;
}

.c0:hover {
  -webkit-transition: color .3s ease;
  transition: color .3s ease;
  background: #007888;
  color: #fff;
}

<button
  className="c0"
  disabled={undefined}
  onClick={[Function]}
  style={undefined}
  type={undefined}
>
  Primary
</button>
`;

exports[`<Button /> snapshots renders selected 1`] = `
.c0 {
  height: 36px;
  min-width: 90px;
  padding: 0 12px;
  border: 0;
  outline: none;
  box-shadow: 0px 0px 2px 2px #8383ac;
  background: #fff;
  color: #000;
  font-size: 0.875em;
  text-transform: uppercase;
  cursor: pointer;
}

.c0:hover {
  -webkit-transition: color .3s ease;
  transition: color .3s ease;
  background: #fafafc;
  color: #3c3c5b;
}

<button
  className="c0"
  disabled={undefined}
  onClick={[Function]}
  style={undefined}
  type={undefined}
>
  Selected
</button>
`;
